### DISCUSSION TITLE
calculate cumulative survival probability?

### DISCUSSION THREAD

**Discussion by gravesti** [vincentarelbundock/marginaleffects#1174] [Q&A]
Thanks for this really nice package! In my own package I have a simple implementation but it would be amazing if I could switch that out to use `marginaleffects` and get support for all these model types.

We are doing discrete time survival modelling with logistic regression and we want to compare the survival probabilities over time. To do this I predict the probability of survival ("response") at each time and take the cumulative product within each id. Then I can average the probabilities at each time within each treatment groups.

I thought it might be possible to insert this `cumprod` step into the marginaleffects workflow. If I try with `predictions()` and a hypothesis function, I don't get the 'id' column, so I don't know which values to aggregate.

Should this be possible?

```r
library(marginaleffects)
library(dplyr)
library(tidyr)
set.seed(2024)
dat <- tibble(
  id = 1:10,
  assigned_treatment = rbinom(10, 1, 0.5),
  x2 = rpois(10, 3),
  outcome_t = rpois(10, lambda = 10 + x2/2 + 3 * assigned_treatment)
) 
dat <- expand_grid(dat, time = 0:15)
dat <- dat |> 
  mutate(
    outcome = if_else(time == outcome_t, 1, 0)
  ) |> 
  filter(time <= outcome_t) |> 
  select(-outcome_t)

my_model <- glm(outcome ~ assigned_treatment + x2 + time,
           data = dat,
           family = binomial)

dg <- datagrid(
  newdata = dat[dat$time == 0,],
  time = 0:10,
  grid_type = "counterfactual"
)

hyp_fun <- function(x, draws) {
  browser() # x doesn't contain 'id' column
  # try to calculate cumprod of predictions here?
  data.frame(term = 1, estimate = 1)
}

predictions(
  model = my_model,
  newdata = dg,
  type = "response",
  variables =  list(assigned_treatment = 0:1),
  hypothesis = hyp_fun
)
```

**Comment by vincentarelbundock** [comment]
This requires some changes to the package, so I'm closing the discussion here in favor of this issue: https://github.com/vincentarelbundock/marginaleffects/issues/1175

Please comment over there.
